### PR TITLE
Additional helpers and improved animation flagging

### PR DESCRIPTION
**Pull request by fallenoak**
* Added indoor flag helper to MOGI chunks
* Broke apart color in MOMT chunks
* Eliminated overhead of static transparency animations (see commit for details)

### CODE REVIEW

File: src/lib/m2/index.js [hunk 1]
```diff
@@ -157,7 +157,11 @@ export default new r.Struct({
 
     this.transparencies.forEach((transparency) => {
       if (transparency.animated) {
-        animated = true;
+        if (transparency.keyframeCount > 1) {
+          animated = true;
+        } else if (transparency.firstKeyframe.value !== 32767) {
+          animated = true;
```
Comment (timkurvers) [inline src/lib/m2/index.js]: I'm wondering whether the transparencies should be decoded as floats instead:

```
transparencies: new Nofs(new AnimationBlock(r.uint16le))
```

It seems to be a similar number as the Quat16Float conversion.

File: src/lib/m2/animation-block.js [hunk 2]
```diff
@@ -53,6 +53,32 @@ export default function(type) {
...
+
+          if (track.keyframes.length > 0) {
+            return track.keyframes[0];
+          }
+        }
+
+        return null;
+      }
+    },
+
```
Comment (timkurvers) [inline src/lib/m2/animation-block.js]: Awesome! :+1:

File: src/lib/m2/index.js (hunk 1 above)
Comment (fallenoak) [inline src/lib/m2/index.js]: They are fixed precision 16-bit floats, I believe. You divide by 32767 to get the number. Any pointers on how to set that up with an animation block?

**Comment by timkurvers** [comment]
:tada:

**Pull request merged**